src/AzureClient/EntryPoint/IEntryPointGenerator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    /// <summary>
+    /// This service is capable of generating entry points for
+    /// job submission to Azure Quantum.
+    /// </summary>
+    public interface IEntryPointGenerator
```

Comment:
In general, IQ# has used dependency injection to make service classes like `EntryPointGenerator` available to their dependents. Best practice with the DI framework we've been using is generally to only depend on interfaces rather than classes, allowing for future extensibility without breaking existing dependencies.

Reply:
Quoting @cgranade's comment, which explains it better than I could have and since it didn't appear as a response to this thread for some reason:

> In general, IQ# has used dependency injection to make service classes like `EntryPointGenerator` available to their dependents. Best practice with the DI framework we've been using is generally to only depend on interfaces rather than classes, allowing for future extensibility without breaking existing dependencies.